### PR TITLE
Revised item appearances

### PR DESCRIPTION
**Pull request by ScipioWright**
Revises how items in the shop and on pedestals are made.
- No longer rotated (so they don't look weird anymore)
- Layers classification icons like the orbs do
- New useful icon (before it was indicated by the lack of an icon)
- Supports multiple classifications being shown at the same time
- Some fixes for death link that I could have sworn I already merged in

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.idea/workspace.xml
+.idea/shelf/Uncommitted_changes_before_Update_at_9_1_2024_3_14_PM__Changes_.xml
+.idea/shelf/Uncommitted_changes_before_Update_at_9_1_2024_3_14_PM_\[Changes]/shelved.patch
```
Comment (heinermann) [inline .gitignore:3]: Just add `.idea/` to gitignore to ignore everything in that directory.

File: data/archipelago/scripts/shopitem_utils.lua [hunk 2]
```diff
@@ -76,20 +76,69 @@ end
...
+				enable_prog_icon = true
+			elseif random_number == 2 then
+				enable_filler_icon = false
+				enable_useful_icon = true
+			end
+		end
 	end
 
 	local item_entity = EntityLoad("data/archipelago/entities/items/" .. item_filename, x, y)
+	if enable_prog_icon == true then
```
Comment (heinermann) [inline data/archipelago/scripts/shopitem_utils.lua:115]: Are these blocks duplicated in 3 files? Seems like most of the logic in this function could go into a utility script? Not a requirement or anything though.

**Comment by ScipioWright** [comment]
Mainly just waiting on player feedback before merging this in. Also going to mess with the progression arrow to make it a little bigger on shop/pedestals

File: data/archipelago/scripts/ap_utils.lua [hunk 3]
```diff
@@ -250,32 +250,75 @@ end
...
 		item_description = "$ap_shopdescription_progression"
-	elseif bit.band(flags, AP.ITEM_FLAG_TRAP) ~= 0 then
+		if enable_useful_icon then
+			item_description = "$ap_shopdescription_proguseful"
+		end
+	end
+
+	if bit.band(flags, AP.ITEM_FLAG_TRAP) ~= 0 then
 		item_filename = "ap_trap_item.xml"
 		item_description = "$ap_shopdescription_trap" .. tostring(Random(1, 8))
```
Comment (ScipioWright) [inline data/archipelago/scripts/ap_utils.lua:275]: Todo: add a way for it to detect if it's useful + trap or progression + trap and give it a different description and give it its actual name so someone doesn't just skip it entirely

**Pull request merged**